### PR TITLE
[release-5.0] OCPBUGS-87208: Adding escapeHAProxySingleQuotes for sanitize

### PR DESCRIPTION
**Pull request by Thealisyed**
This function will allow users to include
certain characters and spacing to avoid
any config injection

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved path rewriting to safely handle single quotes in route paths, preventing configuration errors.

* **Tests**
  * Added comprehensive test coverage for path rewriting functionality, including single quote handling and edge cases.

### CODE REVIEW

File: pkg/router/template/util/rewritetarget/rewritetarget.go [hunk 1]
```diff
@@ -157,3 +157,16 @@ func SanitizeInput(val string) string {
...
+// Bourne shell, and returns the result. Carriage return and line feed
+// characters are removed from the input. The caller is expected to enclose the
+// returned value in single quotes; this function does not itself add any
+// enclosing quote characters.
+func EscapeSingleQuotes(val string) string {
+	val = strings.ReplaceAll(val, `'`, `'\''`)
+	val = strings.ReplaceAll(val, "\n", "")
+	val = strings.ReplaceAll(val, "\r", "")
+	return val
+}
```
[the remark this answers is not in the document]
Comment (Miciah) [reply pkg/router/template/util/rewritetarget/rewritetarget.go:172]: @coderabbitai,

> HAProxy treats single quotes as strong-quote delimiters, and embedded `'` characters can’t be handled by Bourne-shell-style `'\''` inside a single-quoted strong string.

Keep in mind that `val` is expected to be enclosed in single-quotes.  What substituting `'\''` for `'` does in this context is (1) stop quoting the characters of `val`, (2) insert a literal `'` character using `\` to escape it as quoting is no longer active, and (3) start quoting again for the remaining characters of `val`.  

For HAProxy's configuration parsing logic including its handling of quoting and escape codes, please see the definition of `parse_line` in <https://git.haproxy.org/?p=haproxy-2.8.git;a=blob;f=src/tools.c>.

**Comment by Thealisyed** [comment]
infra failure

/retest

**Comment by Miciah** [comment]
@Thealisyed, we need a Jira tracker for the issue in OpenShift 5.0 to link to this PR.

**Comment by Miciah** [comment]
/retitle [release-5.0] OCPBUGS-87208: Adding escapeHAProxySingleQuotes for sanitize

**Comment by rhamini3** [comment]
verified on a 5.0.0 clusterbot cluster

```
% oc get clusterversion
NAME      VERSION                                               AVAILABLE   PROGRESSING   SINCE   STATUS
version   5.0.0-0-2026-06-05-202122-test-ci-ln-lx91xbt-latest   True        False         105m    Cluster version is 5.0.0-0-2026-06-05-202122-test-ci-ln-lx91xbt-latest
% oc create route edge edg-route --service=service-unsecure                                  
route/edg-route created
iamin@iamin-mac openshift-release % oc annotate route edg-route haproxy.router.openshift.io/rewrite-target="/it's" --overwrite 
route.route.openshift.io/edg-route annotated
% oc get route edg-route -o yaml                                                            
apiVersion: route.openshift.io/v1
kind: Route
metadata:
  annotations:
    haproxy.router.openshift.io/rewrite-target: /it's
% oc -n openshift-ingress rsh router-default-56ff885cf7-hzd27
sh-5.1$ cat haproxy.config | grep edg-route -A20
backend be_edge_http:default:edg-route
  mode http
  option redispatch
  option forwardfor
  balance random

  timeout check 5000ms
  http-request add-header X-Forwarded-Host %[req.hdr(host)]
  http-request add-header X-Forwarded-Port %[dst_port]
  http-request add-header X-Forwarded-Proto http if !{ ssl_fc }
  http-request add-header X-Forwarded-Proto https if { ssl_fc }
  http-request add-header X-Forwarded-Proto-Version h2 if { ssl_fc_alpn -i h2 }
  # See the quoting rules in https://tools.ietf.org/html/rfc7239 for IPv6 addresses (v4 addresses get translated to v6 when in hybrid mode)
  acl ipv6_addr src -m sub :
  http-request add-header Forwarded for=\"[%[src]]\";host=%[req.hdr(host)];proto=%[req.hdr(X-Forwarded-Proto)] if ipv6_addr
  http-request add-header Forwarded for=%[src];host=%[req.hdr(host)];proto=%[req.hdr(X-Forwarded-Proto)] if !ipv6_addr
  # Path rewrite target
  http-request replace-path '^(.*)$' '/its\1'
```
/verified by @rhamini3
